### DISCUSSION TITLE
ubuntu docker install adguardhome cannot access webUI

### DISCUSSION THREAD

**Discussion by ww2283** [AdguardTeam/AdGuardHome#5502] [Q&A] [answered]
I thought this would be straightforward but I just cannot access webUI at http://server-ip:3000. Does anyone else experienced things like this? I use docker compose to run adguardhome in Ubuntu 18.04.

```
version: "3.7"

services:

  adguardhome:
    image: adguard/adguardhome
    container_name: adguardhome
    restart: unless-stopped
    network_mode: host
    environment:
      - PUID=1000
      - PGID=1000
      - UMASK=002
      - BIND_HOST=0.0.0.0
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /opt/adguardhome/work:/opt/adguardhome/work
      - /opt/adguardhome/conf:/opt/adguardhome/conf
```

docker logs adguardhome (address part omitted, generally no errors or warnings):
```
2023/02/17 12:17:18.432024 [info] AdGuard Home, version v0.107.24
2023/02/17 12:17:18.432077 [info] This is the first time AdGuard Home is launched
2023/02/17 12:17:18.432081 [info] Checking if AdGuard Home has necessary permissions
2023/02/17 12:17:18.432164 [info] AdGuard Home can bind to port 53
2023/02/17 12:17:18.432168 [info] AdGuard Home updates are disabled
2023/02/17 12:17:18.433168 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db
2023/02/17 12:17:18.434353 [info] auth: initialized.  users:0  sessions:0
2023/02/17 12:17:18.434362 [info] web: initializing
2023/02/17 12:17:18.434369 [info] This is the first launch of AdGuard Home, redirecting everything to /install.html
2023/02/17 12:17:18.434373 [info] AdGuard Home is available at the following addresses:
2023/02/17 12:17:18.435569 [info] go to http://server-ip:3000
```

**Comment by ammnt** [comment]
Did you forget to open the ports?

```
    ports:
       - "3000:3000"
       - "80:80"
       - "443:443"
       - "53:53"
       - "853:853"
       - "754:754"
       - "5443:5443"
       - "67:67"
```
  
etc. More information here:
https://github.com/AdguardTeam/AdGuardHome/wiki/Docker#create-directories-for-persistent-configuration-and-data

**Comment by ww2283** [comment]
but with network_mode: host, any ports specified will be ignored, right? I want adguardhome to take over the network services, so I used network_mode: host. 

I disabled systemd-resolved.service and nginx.service.

To add: if I disable network_mode: host, I can reach webUI by ONLY open 3000:3000 (other ports are being used by caddy; I eventually want to use caddy with adguardhome but for now just to keep it simple and fix the network_mode: host issue). 

further add: `sudo systemctl status docker` shows every docker's running -host-ports except adguardhome. I think this means somehow with network_mode: host, the adguardhome is not taking its port, though `docker logs adguardhome` didn't reveal any trace.

**Comment by ww2283** [comment, marked as answer]
iptables add 3000 port fixed it.

**Reply by rsugumar** [reply]
I have the same issue currently with my mac.  do I've to use pfctl to do the same? Just wondering if there's any security vulnerabilities if I do so?
